contracts/common/Proxy/Proxy.sol

```diff
@@ -3,18 +3,12 @@ import {IERCProxy} from "./IERCProxy.sol";
 import {ProxyStorage} from "./ProxyStorage.sol";
 
 contract Proxy is ProxyStorage, IERCProxy {
```

Comment:
Do not inherit from `ProxyStorage`.
`sload(IMPLEMENTATION_SLOT)` serves the purpose of `proxyTo` var in this proxy pattern.